tests/cupy_tests/core_tests/test_dlpack.py

```diff
@@ -48,19 +48,28 @@ def _get_stream(self, stream_name):
 
     @testing.for_all_dtypes(no_bool=True)
     def test_conversion(self, dtype):
+        orig_array = _gen_array(dtype)
+        out_array = cupy.from_dlpack(orig_array)
+        testing.assert_array_equal(orig_array, out_array)
+        testing.assert_array_equal(
+            orig_array.data.ptr, out_array.data.ptr)
+
+    def test_stream(self):
         allowed_streams = ['null', True]
         if not cuda.runtime.is_hip:
             allowed_streams.append('ptds')
 
         # stream order is automatically established via DLPack protocol
-        for src in allowed_streams:
-            src_s = self._get_stream(src)
-            for dst in allowed_streams:
-                dst_s = self._get_stream(dst)
+        for src_s in [self._get_stream(s) for s in allowed_streams]:
+            for dst_s in [self._get_stream(s) for s in allowed_streams]:
```

Comment:
BTW, how about using `parametrize`
```
@pytest.mark.parametrize("src", ['null', 'ptds', True])
@pytest.mark.parametrize("dst", ['null', 'ptds', True])
```
and `pytest.skip` in `_get_stream`?

Reply:
Looks nice, but seems this doesn't work with unittest-style methods... 😞